application/controllers/admin/Quotas.php

```diff
@@ -58,14 +58,28 @@ private function getData($iSurveyId)
     }
 
     /**
+     * Check permission cfor a survey about quota
+     * @param integer $iSurveyId
      * @param string $sPermission
+     * @param integer|null $iQuotaid
+     * throw Exception
+     * @return void
      */
-    private function checkPermissions($iSurveyId, $sPermission)
+    private function checkPermissions($iSurveyId, $sPermission, $iQuotaid = null)
     {
         if (!empty($sPermission) && !(Permission::model()->hasSurveyPermission($iSurveyId, 'quotas', $sPermission))) {
             Yii::app()->session['flashmessage'] = gT('Access denied!');
             $this->redirectToIndex($iSurveyId);
         }
+        if ($iQuotaid) {
```

Comment:
Is this part of the checking permission? Or this is part of a parent (survey) - child (quota) relationship checking?
Usually these kind of checks are not done on checkPermission routines.
I would say this needs to be moved out of this method.

Reply:
It check permission on quota here. I can rename it to checkValidity , but it still a private function.

Reply:
The permission is checked for the survey. The quota id is only used for checking it belongs to it. right?

Reply:
Yes , and ?

Reply:
Let's unify conversations: https://github.com/LimeSurvey/LimeSurvey/pull/3633#discussion_r1402080740